src/SFML/Window/Unix/Display.cpp

```diff
@@ -104,8 +104,12 @@ std::shared_ptr<_XIM> openXim()
         XSetLocaleModifiers("");
 
         // Create the input context
-        sharedXIM.reset(XOpenIM(UnixDisplayImpl::weakSharedDisplay.lock().get(), nullptr, nullptr, nullptr), XCloseIM);
-        xim = sharedXIM;
+        XIM newXIM = XOpenIM(UnixDisplayImpl::weakSharedDisplay.lock().get(), nullptr, nullptr, nullptr);
+        if (newXIM)
+        {
+            sharedXIM.reset(newXIM, XCloseIM);
+            xim = sharedXIM;
+        }
```

Comment:
```suggestion
        sharedXIM.reset(XOpenIM(UnixDisplayImpl::weakSharedDisplay.lock().get(), nullptr, nullptr, nullptr), XCloseIM);
        if (sharedXIM.reset)
            xim = sharedXIM;
```

Reply:
@ChrisThrasher This would still segfault if I understand the error correctly.
According to the commit message:
> Passing nullptr to XCloseIM segfaults, so check the result of XOpenIM first.

Maybe a simpler solution would be to have a custom deleter that checks for `nullptr` before calling `XCloseIM`, instead of passing directly `XCloseIM` as a custom deleter?

Reply:
Yes that code would still segfault. The issue is doing this is broken:
```
sharedXIM.reset(nullptr, XCloseIM);
```

Because `unique_ptr` always calls the deleter on destruction (even if it contains a `nullptr`) and `XCloseIM` doesn't bother checking if the input is null first:
https://gitlab.freedesktop.org/xorg/lib/libx11/-/blob/master/src/xlibi18n/IMWrap.c#L128

> Maybe a simpler solution would be to have a custom deleter that checks for `nullptr` before calling `XCloseIM`, instead of passing directly `XCloseIM` as a custom deleter?

I think that's a perfectly valid solution as well.

Reply:
> Or maybe not, taking ownership of nullptr is weird.

For some reason I was thinking about `unique_ptr` while writing the above and confusing myself a little...

For `shared_ptr` this is the strange case where `shared_ptr(nullptr)` and `shared_ptr(<pointer which may be null>)` do different things. It's the latter which calls the deleter which is what the problem is here.

Reply:
> Maybe a simpler solution would be to have a custom deleter that checks for nullptr before calling XCloseIM, instead of passing directly XCloseIM as a custom deleter?

I think this is the right solution.

Reply:
I changed the second commit to use a custom deleter instead